File: frontend/src/types/FilmDetail.ts
import Schedule from './Schedule';

export default interface FilmDetail {
    filmId: string,
    title: string,          // Tên phim
    category: string,       // Thể loại
    description: string,    // Mô tả
    director: string|null,  // Đạo diễn
    actors: string|null,    // Diễn viên, các nhau bởi dấu ',' ví dụ: "Nguyễn A,Nguyễn B,Nguyễn B"
    time: number            // Thời gian chiếu
    releaseTime: Date|null,      // Thời gian khởi chiếu, null là không có thời gian chiếu
    country: string,        // Xuất sứ
    poster: string|null,
    trailer: string|null,
    schedules: Array<Schedule>
}

File: backend/Models/FilmDetail.cs
using Cinema.Entities;

namespace Cinema.Models;

public class FilmDetail : Film
{
    public IEnumerable<Schedule> Schedules { get; set; }

    public FilmDetail(Film film)
    {
        FilmId = film.FilmId;
        Title = film.Title;
        Category = film.Category;
        Description = film.Description;
        Director = film.Director;
        Actors = film.Actors;
        Time = film.Time;
        Country = film.Country;
        Poster = film.Poster;
        Trailer = film.Trailer;
        Schedules = new List<Schedule>();
    }

    public FilmDetail(Film film, IEnumerable<Schedule> schedules)
    {
        FilmId = film.FilmId;
        Title = film.Title;
        Category = film.Category;
        Description = film.Description;
        Director = film.Director;
        Actors = film.Actors;
        Time = film.Time;
        Country = film.Country;
        Poster = film.Poster;
        Trailer = film.Trailer;
        Schedules = schedules;
    }
}
